unitTests/utils/AccessibleModalTest.ts

```diff
@@ -0,0 +1,140 @@
+import { $$ } from '../../src/utils/Dom';
+import { Simulate } from '../Simulate';
+import { FocusTrap } from '../../src/ui/FocusTrap/FocusTrap';
+import { KEYBOARD } from '../../src/utils/KeyboardUtils';
+
+export function AccessibleModalTest() {
+  describe('AccessibleModal', () => {
+    let bodyElement: HTMLElement;
+    let titleElement: HTMLElement;
+    let contentElement: HTMLElement;
+    let modalBoxMock: Coveo.ModalBox.ModalBox;
+    let accessibleModal: AccessibleModal;
+    let validationSpy: jasmine.Spy;
+    const modalClass = 'notice-me';
+
+    function createBody() {
+      return (bodyElement = $$('div').el);
+    }
+
+    function createAccessibleModal() {
+      return (accessibleModal = new AccessibleModal(
+        modalClass,
+        createBody() as HTMLBodyElement,
+        (modalBoxMock = Simulate.modalBoxModule())
+      ));
+    }
+
+    function createTitle() {
+      return (titleElement = $$('div').el);
+    }
+
+    function createContent() {
+      return (contentElement = $$('div').el);
+    }
+
+    function createValidationSpy() {
+      (validationSpy = jasmine.createSpy('validation')).and.returnValue(true);
+      return validationSpy as () => boolean;
+    }
+
+    beforeEach(() => {
+      createAccessibleModal();
+    });
+
+    describe('when calling open', () => {
+      let focusTrap: FocusTrap;
+      let container: HTMLElement;
+      let closeButton: HTMLElement;
+      let closeButtonClickSpy: jasmine.Spy;
+
+      beforeEach(() => {
+        accessibleModal.open(createTitle(), createContent(), createValidationSpy());
+        focusTrap = accessibleModal['focusTrap'];
+        container = accessibleModal.element;
+        closeButton = container.querySelector('.coveo-small-close');
+        closeButtonClickSpy = spyOn(closeButton, 'click');
+      });
+
+      it('has an element', () => {
+        expect(accessibleModal.element).toBeTruthy();
+      });
+
+      it('has a wrapper', () => {
+        expect(accessibleModal.wrapper).toBeTruthy();
+      });
+
+      it('has content', () => {
+        expect(accessibleModal.content).toBeTruthy();
+      });
+
+      it("calls ModalBox's open", () => {
+        expect(modalBoxMock.open).toHaveBeenCalledTimes(1);
+      });
+
+      it("doesn't call the validation function", () => {
+        expect(validationSpy).not.toHaveBeenCalled();
+      });
+
+      it('creates a focus trap', () => {
+        expect(focusTrap).not.toBeNull();
+      });
+
+      it("doesn't disable its focus trap", () => {
+        expect(focusTrap['enabled']).toEqual(true);
+      });
+
+      it('sets aria-modal to true on the modal element', () => {
+        expect(container.getAttribute('aria-modal')).toEqual('true');
+      });
+
+      it('gives the close button a label', () => {
+        expect(closeButton.getAttribute('aria-label')).toBeTruthy();
+      });
+
+      it('gives the close button a tabindex', () => {
+        expect(closeButton.tabIndex >= 0).toEqual(true);
```

Comment:
Could we be more strict with the assertion on the tabIndex? Is it always set to `0` or does it change?

Reply:
It gives it a `tabIndex` and doesn't change it afterwards. This is an accessibility property that is missing in the `modalbox` package.